### ISSUE TITLE
Clarify Project name and path

### ISSUE THREAD

**Issue by elburz** [openframeworks/projectGenerator#139]
Context: Have experience with oF, use VS2015 with the oF plugin normally , new to using the project generator.

Problem: I find the project name and path combination confusing

Questions I ask myself that I can't easily answer:
- Can project path be any path anywhere?
- Should project path stick to the general practice of being ../../../ away from the root? 
- Will project name generate a new folder for me as well (in which case do I have to point it to ../../ away from root)? Or will it just dump files into a folder I make and then point the project path at that folder? 

Possible solutions:
- A single line below each field name that describes what is expected/best practice and what the tool will do for you. i.e. "Project name will create a folder and place your new project in it."
- If tooltips aren't hard, maybe a small "?" icon next to the names that will have a sentence or two description of expected/best practice
- Depending on how much a standard design process wants to be created in oF, maybe project path should already lock in/grey out the path to your "myApps" folder using the oF path in settings and only let you add a project name, which then creates the folder for you. Then there could be a possible "Expert mode" button in the settings that allows you to do whatever you like with your project path.

**Comment by ofZach** [comment]
**Can project path be any path anywhere?**

yes, but keep in mind that if it's outside of the OF folder, the project may use fixed paths, so that if you move the project or OF the paths may break.  If it's inside of the OF folder, the paths will be relative.   

**Should project path stick to the general practice of being ../../../ away from the root?**

any distance away from the root should be fine...  ie, ../../ vs ../../../../ is ok. 

**Will project name generate a new folder for me as well (in which case do I have to point it to ../../ away from root)? Or will it just dump files into a folder I make and then point the project path at that folder?**

not sure I undestand this question.   if you make project name blah, with path:

/Users/zachlieberman/Desktop/of_v0.9.3_osx_release/apps/myApps

it will make a folder called: 

/Users/zachlieberman/Desktop/of_v0.9.3_osx_release/apps/myApps/blah

with a project (xcode project file) called blah inside it that makes a bin named blah.

**Comment by elburz** [comment]
Makes sense, some of that is new information to me. Re: the last question that was a bit convoluted, I was asking if instead of what you mentioned it did, if it did something like:

I would make a folder called:
/Users/zachlieberman/Desktop/of_v0.9.3_osx_release/apps/myApps/blah

Then I'd set project name to blah. I would expect it to dump project files in:
/Users/zachlieberman/Desktop/of_v0.9.3_osx_release/apps/myApps/blah

and not:
/Users/zachlieberman/Desktop/of_v0.9.3_osx_release/apps/myApps/blah/blah

I guess my overall concern/confusion is what would be your suggested/preferred design path? Like if this was the first thing someone was using in oF was the project generator, what would you want them to do. I don't think that direction/best practice is present currently, which for some people may be ok, but I think for some people it could lead to confusion and frustration. I think the solutions I mentioned above, any of them could work. Thoughts?

**Comment by arturoc** [comment]
i've always found the separation between path and project name confusing too but then most ides do the same or even more convoluted. perhaps something that would make it clearer would be to check if the name of the project is the same as the last part of the path and in that case don't create the extra folder. I actually thought we were already doing that

**Comment by ofZach** [comment]
I think maybe we can simply rename "path" to something like "containing folder" ? I think it's helpful to think of path more like that -- a folder which contains your project -- rather than to add auto smarts (I can think of instances where I might want the containing folder and the project folder be the same, ie, blah/blah/blah.xcproj)

**Comment by arturoc** [comment]
makes sense, also what @elburz suggested initially, having some kind of help tooltip that shows the short explanations you've posted here would be really helpful
